src/screens/QuestionScreen/QuestionScreen.js

```diff
@@ -0,0 +1,167 @@
+import { LOGO } from '../../config/styles.js';
+import styles from './styles';
+import moment from 'moment';
+import Icon from 'react-native-vector-icons/Ionicons';
+import ImagePicker from 'react-native-image-crop-picker';
+import * as Progress from 'react-native-progress';
+
+class QuestionScreen extends  React.Component {
+    state = {queryInput : '', setImage : null, setDisplayImage: null, transferred: 0};
+
+    takePhotoFromLib = () => {
+        const {setImage} = this.state
+        this.setState({transferred: 0})
+        ImagePicker.openPicker({
+          width: 300,
+          height: 300,
+          cropping: true,
+        }).then((image) => {
+          const imageUri = image.path;
+          this.setState({setImage: imageUri});
+        }).catch("Unknown Error Occured")
+      };
+
+      uploadImage = async () => {
+        const {setImage} = this.state
+        if(setImage == null){
+          return
+        }
+          const uri = setImage;
+          const childPath = `post/${firebase.auth().currentUser.uid}/${Math.random().toString(36)}`;
+  
+          const response = await fetch(uri);
+          const blob = await response.blob();
+  
+          const task = firebase
+              .storage()
+              .ref()
+              .child(childPath)
+              .put(blob);
+
+          const taskCompleted = () => {
+              const {transferred} = this.state
+              task.snapshot.ref.getDownloadURL().then((snapshot) => {
+                  this.setState({setDisplayImage: snapshot})
+              }).catch((error) => {
+                alert(error)
+              })
+          }
+  
+          const taskError = snapshot => {
+            alert("Error Occured")
+        }
+  
+        task.on("state_changed", snapshot => {
+            const {transferred} = this.state
+            this.setState({ transferred: Math.round(snapshot.bytesTransferred / snapshot.totalBytes) * 10000})
+          }, taskError, taskCompleted);
+      }
+
+      showAlert = () => {
+          const {setImage} = this.state
+          const {setDisplayImage} = this.state
+        Alert.alert(  
+            'Alert Title',  
+            'My Alert Msg',  
+            [  
+                {  
+                    text: 'Remove',  
+                    onPress: () => {this.setState({
+                        setDisplayImage: null,
+                        setImage: null
+                    })},  
+                    style: 'cancel',  
+                },  
+                {text: 'Show', onPress: () => Linking.openURL(setDisplayImage)},  
+            ]  
+        );  
+      }
+
+    addQuery = () => {
+        const {setDisplayImage, setImage, queryInput} = this.state
+        var timeDate = moment()
+        const Useruid = firebase.auth().currentUser.uid
+        if(queryInput && queryInput.length > 0){
+            firebase.firestore().collection("queries").add({
+                id: Useruid,
+                queryInput,
```

Comment:
Isn't there should be a key here?

Reply:
Do you mean key-value pair? Using this like `queryInput: queryInput` is same as `queryInput` I guess. The data added through this line gives me the same output as with key: value pair. Let me know if this approach is fine or should I change it?

Reply:
Can you debug and see what is the value of the passing queryInput? I'm not confident on this approach however. If you say this is working, that's fine. But better to verify with debugging the value.

Reply:
I double-checked it using the remote debugger, It has the same value that I passed in using textinput. I guess this may lead to confusion in the future so I should add a key to it!